99553: Port over necessary files for pdf filling

balexandr opened the pull request:
Bring the pdf_fill library for burials into the module. utilize/enable the register_form function in burials/engine similar to pensions


## Summary

- Update burial engine file to register burial form in initialize
- Port over burial specific files and properly namespace
- Remove old references to burial forms in `filler.rb`
- Port over tests (has about 95%+ coverage)

## Related issue(s)

- department-of-veterans-affairs/va.gov-team#99553

## Testing done

- [x] Manual form submission WITH `burial_module_enabled`
- [x] Manual form submission WITHOUT `burial_module_enabled`
- [x] Rspec

## Screenshots
<img width="698" alt="Screenshot 2025-01-28 at 3 12 57 PM" src="https://github.com/user-attachments/assets/fae8b4c2-ca3c-4fb1-8dab-936d0b5c58ce" />


## What areas of the site does it impact?
Burial form 21P-530EZ

TaiWilkin reviewed (dismissed):
This is working well for me, looks good

balexandr commented on app/sidekiq/lighthouse/submit_benefits_intake_claim.rb:
```diff
@@ -146,11 +146,19 @@ def stamped_pdf_with_form(form_id, path, timestamp)
         page_number: 5,
         size: 9,
-        template: "lib/pdf_fill/forms/pdfs/#{form_id}.pdf",
+        template: template_path(form_id),
         multistamp: true
       )
     end
 
+    def template_path(form_id)
+      if form_id == '21P-530EZ'
+        "modules/burials/lib/pdf_fill/forms/pdfs/#{form_id}.pdf"
+      else
+        "lib/pdf_fill/forms/pdfs/#{form_id}.pdf"
+      end
+    end
+
```
@wayne-weibel I forgot, this is only called for Burials anyways

rmtolmach commented on modules/burials/spec/fixtures/pdf_fill/21P-530EZ/overflow.json:
```diff
@@ -65,11 +65,11 @@
   "burialAllowance": false,
   "plotAllowance": false,
-  "amountGovtContribution": 9999999999999999999,
+  "amountGovtContribution": " 9999999999999999999",
```
Is that extra space at the beginning of the string on purpose?

balexandr replied:
@rmtolmach yea it does a rshift. If you look at the kitchen_sink or simple you'll see it

rmtolmach reviewed (commented):
Is there any way to split this PR up? I'm getting a bit lost trying to figure out what's brand new vs. moved.

balexandr commented:
> Is there any way to split this PR up? I'm getting a bit lost trying to figure out what's brand new vs. moved.

@rmtolmach Not really, everything relies on each other. I ported over all PDF related things into our module and deleted it from the public lib folder and then I had to update all the tests. Originally the PR was much smaller but tests kept failing

rmtolmach reviewed (approved):
I understand you can't always split things up. LGTM.
